### PR TITLE
Add scripts for importing and stopping MariaDB data

### PR DESCRIPTION
**Pull request by stephdl**
On slower machines, MariaDB initialization may exceed the systemd timeout, leading to errors. To prevent this, MariaDB is now initialized separately before starting the systemd service, avoiding the timeout issue.

A flag file (`mariadb-init-done`) is created at the end of the initialization to confirm it has completed successfully, ensuring it only runs once. This flag has also been incorporated into backup, clone, and update actions to maintain compatibility with previous versions.

https://github.com/NethServer/dev/issues/7078

### CODE REVIEW

**Review by Amygos — CHANGES_REQUESTED** [review]
All the tests are failing at the install module step. The journal of the `create-module`:
```
Oct 22 10:09:05 example.leader.example-domain.net agent@nethvoice11[459522]: error: 'Can't connect to local server through socket '/run/mysqld/mysqld.sock' (2)'
Oct 22 10:09:05 example.leader.example-domain.net agent@nethvoice11[459522]: Check that mariadbd is running and that the socket: '/run/mysqld/mysqld.sock' exists!
Oct 22 10:09:05 example.leader.example-domain.net agent@nethvoice11[459522]: 2024-10-22 10:09:05+00:00 [ERROR] [Entrypoint]: Unable to shut down server.
Oct 22 10:09:05 example.leader.example-domain.net systemd[459502]: libpod-b6b352be9e03eb2018823e1223dd6a65c2aad57f4b2f69b9c55179a17c34f449.scope: Consumed 5.471s CPU time.
Oct 22 10:09:05 example.leader.example-domain.net agent@nethvoice11[459522]: task/module/nethvoice11/aac5c9fe-0638-4222-973e-92ce05d2e964: action "create-module" status is "aborted" (1) at step 50import-mariadb
Oct 22 10:09:05 example.leader.example-domain.net agent@cluster[455727]: Assertion failed
Oct 22 10:09:05 example.leader.example-domain.net agent@cluster[455727]:   File "/var/lib/nethserver/cluster/actions/add-module/50update", line 223, in <module>
Oct 22 10:09:05 example.leader.example-domain.net agent@cluster[455727]:     agent.assert_exp(create_module_result['exit_code'] == 0) # Ensure create-module is successful
Oct 22 10:09:05 example.leader.example-domain.net agent@cluster[455727]: task/cluster/7895e6d8-e811-415a-a45e-10bc24897893: action "add-module" status is "aborted" (2) at step 50update
```
The action fails because the shutdown of the database in the file [mariadb/docker-entrypoint-initdb.d/99_stop_mariadb_import.sh](https://github.com/nethesis/ns8-nethvoice/pull/324/files#diff-d99eedea525c5258538126af63c401ce1b88ce224d693adf64d744ef67cd2460) conflicts with the shutdown made by the container [entrypoint script](https://github.com/MariaDB/mariadb-docker/blob/cd10ba6c780e49e732ca0eacb76affaac6e87d74/10.8/docker-entrypoint.sh#L148).

**Comment by Amygos** [comment]
We can use this approach to allow the container to start, perform only the initialization phase, and then gracefully exit.
https://github.com/docker-library/mysql/pull/423#issuecomment-562355462

**Comment by stephdl** [comment]
it should be good @Amygos

**Review by Stell0 — APPROVED** [review]

**Review by Amygos — CHANGES_REQUESTED** [review]

File: imageroot/actions/create-module/50import-mariadb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    --env=CDRDBHOST \
+    --env=CTIUSER \
+    --env=NETHCTI_DB_USER \
+    --env=PHONEBOOK* \
+    --env=ASTERISK_RTPSTART \
+    --env=ASTERISK_RTPEND \
+    --env=ASTERISK_SIP_PORT \
+    --env=ASTERISK_SIPS_PORT \
+    --env=ASTERISK_IAX_PORT \
+    --env=ASTMANAGERPORT \
```
[the remark this answers is not in the document]
Comment (stephdl) [reply imageroot/actions/create-module/50import-mariadb]: I got a failure and probably I have been added, as you can see we could either add the file or add each variables

File: imageroot/actions/create-module/50import-mariadb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# Copyright (C) 2024 Nethesis S.r.l.
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+
+# Import MariaDB data
+
+podman run \
+    --rm \
+    --replace \
+    --name=mariadb-import \
```
[the remark this answers is not in the document]
Comment (Amygos) [reply imageroot/actions/create-module/50import-mariadb]: The file names aren't fixed.

[the remark this answers is not in the document]
Comment (stephdl) [reply imageroot/actions/create-module/50import-mariadb]: done

File: imageroot/actions/create-module/50mariadb-init
[the remark this answers is not in the document]
Comment (Amygos) [reply imageroot/actions/create-module/50mariadb-init:1]: For clone-module, and restore-module actions, it is not a problem, but I am not sure about import-module.
@stephdl Have you checked it?

[the remark this answers is not in the document]
Comment (stephdl) [reply imageroot/actions/create-module/50mariadb-init:1]: restore-module we have an issue

`Oct 23 10:34:08 R2-pve agent@nethvoice2[55974]: [00] 2024-10-23 08:34:08 Original data directory /var/lib/mysql is not empty!`

I think the safer approach is to init the mariadb during the configure module maybe, but we have to test if the volume exists or not

what do you think

**Comment by stephdl** [comment]
closed https://github.com/nethesis/ns8-nethvoice/pull/326

**Pull request closed without merging**